subsys/console/getchar.c

```diff
@@ -117,3 +120,25 @@ void console_init(void)
 	uart_irq_callback_set(uart_dev, uart_isr);
 	uart_irq_rx_enable(uart_dev);
 }
+
+#elif defined(CONFIG_SOC_POSIX)
+
+#include <stdio.h>
+
+int console_putchar(char c)
+{
+	putchar(c);
```

Comment:
Again assuming that everybody who uses this soc wants to route directly to libc's putchar is a very wild  assumption (which certainly does not hold in my other boards)